api/samples/inscount.cpp

```diff
@@ -164,9 +164,38 @@ event_bb_analysis(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
             }
         }
     }
-    /* Count instructions */
-    for (instr = instrlist_first_app(bb), num_instrs = 0; instr != NULL;
-         instr = instr_get_next_app(instr)) {
+
+    /* Count instructions. If an emulation client is running with this client,
+     * we want to count all the native instructions and the emulated instruction
+     * but NOT the native instructions used for emulation.
+     */
+    bool is_emulation = false;
+    for (instr = instrlist_first(bb), num_instrs = 0; instr != NULL;
+         instr = instr_get_next(instr)) {
```

Comment:
I agree a separate PR would be better. I will check drutil_expand_rep_string_ex() tonight. Just to be clear, am I looking at the implementation to check it is safe to expand within a series of emulation instructions with the start/stop emulation labels remaining correctly placed?

Reply:
Just looking for anything that might not work: labels around the entire instrlist somehow causing an issue, branches inside the label region being a problem.  I think it will all be fine so I will retract my request -- even if we hit something unanticipated when implementing the rep string we can go back and change the interface if it hasn't been too long since committing it and we didn't do an official release in between (though those seem to be pretty rare these days).